data-raw/dtc_formats.R

```diff
@@ -0,0 +1,27 @@
+## code to prepare `dtc_formats` dataset goes here
+
+dtc_formats <- tibble::tribble(
+  ~fmt, ~type, ~description,
+  "ymd", "date", "Parses a date: year, month, and month day.",
+  "y m d", "date", "Parses a date: year, month, and month day.",
+  "y-m-d", "date", "Parses a date: year, month, and month day.",
+  "dmy", "date", "Parses a date: month day, month and year.",
+  "d m y", "date", "Parses a date: month day, month and year.",
+  "d-m-y", "date", "Parses a date: month day, month and year.",
+  "ym", "date", "Parses a date: year and month.",
+  "y m", "date", "Parses a date: year and month.",
+  "y-m", "date", "Parses a date: year and month.",
+  "my", "date", "Parses a date: month and year.",
+  "m y", "date", "Parses a date: month and year.",
+  "m-y", "date", "Parses a date: month and year.",
```

Comment:
Can we also add the below date and time formats? `-` indicates an unknown component in it.

dd MMM yyyy HH nn ss
HH nn
HH:nn:ss
MMM-
HH nn ss
dd MMM yyyy HH mm ss
dd- MMM- yyyy
dd MMM yyyy HH:nn:ss

HH:nn
dd MMM yyyy hh:nn:ss:rr
dd- MMM- yyyy-
dd MMM yyyy hh nn ss
dd MMM yyyy hh:nn:ss
MMM- yyyy
dd- MMM- yyyy hh:mm:ss
dd- MMM- yyyy HH nn ss
dd- MMM- yyyy- HH nn
dd- MMM yyyy
dd MMM yyyy
dd MMM yyyy HH:nn
hh:nn
yy:mm
yyyy
nn ss
HH mm
dd MMM yyyy hh ss
dd MMM yyyy HH nn
MMM yyyy
HH+ nn

if we do not want to add them, we need to find a way for this function to process it. Currently, I am not able to provide dd MMM yyyy, it gives me an error.

Reply:
So `"MMM-"` means month, including unknown values? But what forms can these unknown values take?

Reply:
The `"rr"` in `"dd MMM yyyy hh:nn:ss:rr"` was not in the original specification request as this is an additional time component beyond hours, minutes and seconds. Is this meant to be fractional seconds?